### PR TITLE
ROC-4625 - repaymentPlan undefined error

### PR DESCRIPTION
**Pull request by rattur**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-4625

### Change description

Fix for repayment plan undefined error

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x ] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/main/features/claimant-response/routes/counter-offer-accepted.ts [hunk 1]
```diff
@@ -34,10 +34,16 @@ export default express.Router()
       const isCourtOrderPaymentPlanConvertedByDefendantFrequency =
         (defendantPaymentOption !== 'BY_SPECIFIED_DATE') ? defendantPaymentPlan.frequency && differentPaymentFrequency : false
 
+      let courtOrderPaymentPlan
+
+      if (draft.document.courtOfferedPaymentIntention !== undefined) {
```
Comment (andrewwa-kainos) [inline src/main/features/claimant-response/routes/counter-offer-accepted.ts]: Could still be `null`.
Suggest either `if (draft.document.courtOfferedPaymentIntention) {` or just using the ternary boolean operator for the options `courtOrderPaymentPlan: draft.document.courtOfferedPaymentIntention ? draft.document.courtOfferedPaymentIntention.repaymentPlan : undefined`

Comment (rattur) [reply src/main/features/claimant-response/routes/counter-offer-accepted.ts]: Yeah ok

File: src/main/features/claimant-response/routes/counter-offer-accepted.ts [hunk 2]
```diff
@@ -34,10 +34,16 @@ export default express.Router()
...
+        courtOrderPaymentPlan = draft.document.courtOfferedPaymentIntention.repaymentPlan
+      }
+
       res.render(Paths.counterOfferAcceptedPage.associatedView, {
         isCourtOrderPaymentPlanConvertedByDefendantFrequency: isCourtOrderPaymentPlanConvertedByDefendantFrequency,
         claimantPaymentPlan: claimantPaymentPlan,
-        courtOrderPaymentPlan: draft.document.courtOfferedPaymentIntention.repaymentPlan
```
Comment (sabahirfan) [inline src/main/features/claimant-response/routes/counter-offer-accepted.ts]:
```
courtOrderPaymentPlan: draft.document.courtOfferedPaymentIntention ? draft.document.courtOfferedPaymentIntention.repaymentPlan : undefined 
```

File: src/main/features/claimant-response/views/counter-offer-accepted.njk [hunk 3]
```diff
@@ -20,13 +20,15 @@
               courtOrderFrequency: courtOrderPaymentPlan.paymentSchedule | renderPaymentSchedule
```
Comment (sabahirfan) [inline src/main/features/claimant-response/views/counter-offer-accepted.njk]: Can't this be null/undefined here? you have a check on line 23 but not here.

Comment (rattur) [reply src/main/features/claimant-response/views/counter-offer-accepted.njk]: Potentially yes, well spotted

**Comment by rattur** [comment]
This issue no longer exists therefore PR is closed

**Comment by rattur** [comment]
This issue no longer exists therefore PR is closed

**Pull request closed without merging**